### PR TITLE
Convert from Microsoft.Azure.Storage.Blob to new Azure.Storage.Blobs

### PR DESCRIPTION
**Pull request by MattGal**
Per the folks on the Azure storage SDK and server-side teams, this is an attempt to address https://github.com/dotnet/core-eng/issues/8582 as the new libraries are ostensibly more thread-safe.

I ran this build to validate it: https://dnceng.visualstudio.com/internal/_build/results?buildId=516665&view=logs&j=f7bb3748-e35d-53de-9dd7-460ecd2014d6&t=299a6d6d-a279-539e-b6e9-1576018947a8 but it's clear that this doesn't exercise the same call stack from the above issue, so I'm reaching out to some runtime folks to make sure I can try out their build(s). 

Relevant runtime build testing this package: https://dnceng.visualstudio.com/internal/_build/results?buildId=516821&view=results

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 1]
```diff
@@ -1,10 +1,9 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using Microsoft.Azure.Storage;
-using Microsoft.Azure.Storage.Auth;
-using Microsoft.Azure.Storage.Blob;
+using Azure;
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: Missing copyright header.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 2]
```diff
@@ -57,131 +59,127 @@ public static string CalculateMD5(string filename)
...
 
         /// <summary>
-        /// Return a bool indicating whether a local file content is same as 
-        /// the content of the pointed blob. If the blob has the ContentMD5
-        /// property set the comparison is performed uniquely using that.
-        /// Otherwise a byte-per-byte comparison with the content of the file
-        /// is performed.
+        /// Return a bool indicating whether a local file content is the same as 
+        /// the content of a given blob. If the blob has the ContentHash
+        /// property set (all blobs should) the comparison is performed using that (MD5 hash)
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: > (all blobs should)

Might be useful to point out why this is important to handle, then--I think old clients didn't publish the hash?

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 3]
```diff
@@ -57,131 +59,127 @@ public static string CalculateMD5(string filename)
...
+                var blobMD5 = Convert.ToBase64String(properties.ContentHash);
                 return blobMD5.Equals(localMD5, StringComparison.OrdinalIgnoreCase);
             }
             else
             {
-                int OneMegaBytes = 1 * 1024 * 1024;
-
-                if (blobReference.Properties.Length < OneMegaBytes) {
-                    byte[] existingBytes = new byte[blobReference.Properties.Length];
+                int oneMegabyte = 1 * 1024 * 1024;
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: I think it'd make it a *little* clearer when it shows up later to name it like a conversion, `bytesPerMegabyte`.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs (hunk 2 above)
Comment (MattGal) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: That's correct, evidently the "old clients" here are < 2012.

**Comment by MattGal** [comment]
[Tested pipeline](https://dev.azure.com/dnceng/internal/_build/results?buildId=516821&view=logs&j=f7bb3748-e35d-53de-9dd7-460ecd2014d6&t=299a6d6d-a279-539e-b6e9-1576018947a8) was totally green through publishing, so I will merge.

**Pull request merged**